sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -0,0 +1,232 @@
+package com.mapbox.maps
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.FrameLayout
+import androidx.annotation.LayoutRes
+import androidx.annotation.VisibleForTesting
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.bindgen.Expected
+import com.mapbox.maps.viewannotation.ViewAnnotation
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.collections.HashMap
+
+internal class ViewAnnotationManagerImpl(
+  private val mapView: MapView
+) : ViewAnnotationManager, ViewAnnotationPositionsUpdateListener {
+
+  private val mapboxMap: MapboxMap = mapView.getMapboxMap()
```

Comment:
mapboxMap seems to be a hidden dependency of ViewAnnotationManagerImpl. 
Should it be a constructor arg instead? Should make testing also easier

Reply:
can't get what hidden dependency is here tbh, in our terms `MapView` almost ~ to `MapboxMap`. Also tests are already written and I did not see this as an issue tbh 😄 
can you add more clarity here?

Reply:
:shipit:

Reply:
IMO no difference - could be mocked via `mapView` instead